deps/pathfinder/Dockerfile

```diff
@@ -0,0 +1,57 @@
+# Stage 1: Build the pathfinder binary
+FROM rust:slim-bookworm AS builder
+
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl \
+    git \
+    build-essential \
+    libssl-dev \
+    pkg-config \
+    unzip \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install protobuf (specific version)
+RUN curl -LO https://github.com/protocolbuffers/protobuf/releases/download/v28.3/protoc-28.3-linux-x86_64.zip && \
+    unzip protoc-28.3-linux-x86_64.zip -d /usr/local && \
+    chmod 755 /usr/local/bin/protoc && \
+    rm protoc-28.3-linux-x86_64.zip && \
+    protoc --version
+
+
+# Clone and build Pathfinder
+RUN git clone https://github.com/Moonsong-Labs/pathfinder.git && \
+    cd pathfinder && \
+    git checkout notlesh/snos-requirements-2024-12-30 && \
+    cargo build --release --bin pathfinder
```

Comment:
This is also ugly, we should be using a non-custom version of Pathfinder. 
Also please add to improvement ticket

Reply:
Already added to the list! Besides, we should move to Docker images instead of building the stuff by ourselves